### PR TITLE
Refactoring Sum of binary numbers & incorrect time complexity of square root.

### PR DESCRIPTION
**Pull request by ajinkyakolhe112**
Refactored code of two binary number summations.
Less code replication.

&
Elaborated on complexity of square root. Not same as binary search, though binary search is used to find closest number.

### CODE REVIEW

**Comment by ajinkyakolhe112** [comment]
changelist https://github.com/ajinkyakolhe112/Algorithms/commit/af25d4206fb0f82ad3b353f6f3db33127a4cb800, shouldn't cause any failure, it includes change in commented part.

**Comment by ajinkyakolhe112** [comment]
Hi,
are changes not acceptable? Please let me know if you have time. Thanks.

File: src/main/java/com/github/pedrovgs/problem3/SumBinaryNumbers.java [hunk 1]
```diff
@@ -9,126 +9,111 @@
...
+     * To solve this problem we are using 'Integer.parseInt(number,base)' method which complexity
+     * order is linear O(N) two times. Then, we are going to calculate the sum of this parsed data
+     * and use 'Integer.toBinaryString' which complexity order is also O(N). In both cases, N
+     * represents the number of digits of input number in binary representation.
+     * <p/>
+     * This implementation is working and is valid but we have to take into the execution time used
+     * to calculate this sum.
+     */
+    public String sumBinaryNumbersCheating(String n1, String n2) {
+        validateInput(n1, n2);
```
Comment (pedrovgs) [inline src/main/java/com/github/pedrovgs/problem3/SumBinaryNumbers.java]: Why did you introduced all these spaces and the ``<p/>`` element?

**Comment by pedrovgs** [comment]
Thanks for your pull request @ajinkyakolhe112!! The code is ok, but please review the checkstyle failures and the ``<p/>`` characters and whitespace you've added to the javadoc.

Here you have the pull request log. https://travis-ci.org/pedrovgs/Algorithms/builds/107573993

**Comment by ajinkyakolhe112** [comment]
Closing pull request. Fixed the checkstyle error. Creating a new pull request

File: src/main/java/com/github/pedrovgs/problem3/SumBinaryNumbers.java (hunk 1 above)
Comment (ajinkyakolhe112) [inline src/main/java/com/github/pedrovgs/problem3/SumBinaryNumbers.java]: My Auto format setting had that option checked. I unchecked it. Sorry.

**Pull request closed without merging**